packages/multichain/src/adapters/caip-permission-adapter-permittedChains.ts

```diff
@@ -0,0 +1,136 @@
+import { toHex } from '@metamask/controller-utils';
+import type { Hex } from '@metamask/utils';
+import { KnownCaipNamespace } from '@metamask/utils';
+
+import type { Caip25CaveatValue } from '../caip25Permission';
+import {
+  KnownNotifications,
+  KnownRpcMethods,
+  KnownWalletScopeString,
+} from '../scope/constants';
+import { getUniqueArrayItems, mergeScopes } from '../scope/transform';
+import type { InternalScopesObject, InternalScopeString } from '../scope/types';
+import { parseScopeString } from '../scope/types';
+
+/**
+ * Gets the Ethereum (EIP155 namespaced) chainIDs from the required and optional scopes.
+ * @param caip25CaveatValue - The CAIP-25 caveat value from which to get the Ethereum chainIDs.
+ * @returns An array of Ethereum chainIDs.
+ */
+export const getPermittedEthChainIds = (
+  caip25CaveatValue: Pick<
+    Caip25CaveatValue,
+    'requiredScopes' | 'optionalScopes'
+  >,
+) => {
+  const ethChainIds: Hex[] = [];
+  const sessionScopes = mergeScopes(
+    caip25CaveatValue.requiredScopes,
+    caip25CaveatValue.optionalScopes,
+  );
+
+  Object.keys(sessionScopes).forEach((scopeString) => {
+    const { namespace, reference } = parseScopeString(scopeString);
+    if (namespace === KnownCaipNamespace.Eip155 && reference) {
+      ethChainIds.push(toHex(reference));
+    }
+  });
+
+  return getUniqueArrayItems(ethChainIds);
+};
+
+/**
+ * Adds an Ethereum (EIP155 namespaced) chainID to the required and optional scopes.
```

Comment:
This doesn't seem to do as described. If the chain ID is found on either required or optional scopes, the caveat is returned unchanged.

Also, if it's not found, it's only added on the optional scope.

Reply:
Correct. Will update the TSDoc accordingly

Reply:
https://github.com/MetaMask/core/pull/4784/commits/aba885783234b4daead5da592e248a28d773b226